riverdriver/riverpgxv5/river_pgx_v5_driver.go

```diff
@@ -507,31 +507,31 @@ func (l *Listener) Connect(ctx context.Context) error {
 }
 
 func (l *Listener) Listen(ctx context.Context, topic string) error {
-	l.mu.RLock()
-	defer l.mu.RUnlock()
+	l.mu.Lock()
+	defer l.mu.Unlock()
```

Comment:
I missed it before until adding more tests to stress it, but even `LISTEN` keeps the connection busy, so every listener operation should actually take a full mutex lock instead of allowing multiple readers.